lib/tasks/grade.rake

```diff
@@ -128,37 +132,60 @@ namespace :grade do
     end
     puts "Grade token has been reset successfully."
   end
+end
+
+def sync_specs_with_source(full_reponame, remote_sha, repo_url)
+  # Discard unstaged changes in spec folder
+  `git checkout spec -q`
+  `git clean spec -f -q`
+  local_sha = `git ls-tree HEAD #{project_root.join("spec")}`.chomp.split[2]
+
+  unless remote_sha == local_sha
+    files_and_subfolders_inside_specs = Dir.glob("spec/*")
+    # Remove the contents of the directory
+    FileUtils.rm_rf(files_and_subfolders_inside_specs)
 
+    find_or_create_directory("tmp")
+    download_file(repo_url, "tmp/spec.zip")
+    extracted_zip_folder = extract_zip("tmp/spec.zip", "tmp")
+    source_directory = extracted_zip_folder.join("spec")
+    overwrite_spec_folder(source_directory)
+
+    FileUtils.rm(project_root.join("tmp/spec.zip"))
+    FileUtils.rm_rf(extracted_zip_folder)
+    `git add spec/`
+    `git commit spec/ -m "Update project tests" --author "First Draft <grades@firstdraft.com>"`
```

Comment:
```suggestion
    `git commit spec/ -m "Update spec folder to match official version" --author "First Draft <grades@firstdraft.com>"`
```

Or maybe something like that? Just to make it explicit via the commit message to students that this was caused by them attempting to edit the spec/ folder? I hesitate to use a word like "upstream" even though that is correct, as it might not be clear to beginners.

Reply:
Related, this message on the commit is a little curious. Are we sure we want to use that `grades@firstdraft.com` email?

![image](https://github.com/user-attachments/assets/7a2b11d2-4dc3-4ca4-a1b5-acd5543ed8d8)

Reply:
huh, interesting. I didn't think that was an email GitHub would recognize since there isn't a GitHub account with that email... 🤔

My intention was to have "First Draft" make the commit so it's clear to the student who made the commit. As long as it doesn't break anything, I think it's fine for now.

Reply:
I think it looks fine with that email and on second though it kind of makes sense (the commit isn't from them, and GitHub is letting them know; and the grades email makes it clear it's coming from them running the `rake grade` command and receiving some updates). 

My other point about the more informative commit message still stands though.